### PR TITLE
Fixing readGWLdata to work with new data

### PR DESCRIPTION
**Pull request by steffilazerte**

### CODE REVIEW

**Comment by steffilazerte** [comment]
This pull does the following:
- adds a new function `get_gwl` to download well data (including documentation)
- adds tests to ensure the working of downstream functions
- updates the vignette to use `get_gwl`
- removes most global variables (but see comments on issue #1)

**Comment by steffilazerte** [comment]
Also forgot to mention: The only thing missing from the downloaded/formatted data is two additional metadata fields: EMS_ID and Station_Name. Right now they're filled with NA, and I'm not sure where to get the information from without web-scraping the well details ([for example](https://a100.gov.bc.ca/pub/wells/wellsreport1.do?wellTagNumber=309), but even this one doesn't have an EMS ID...).

**Comment by ateucher** [comment]
Thanks @steffilazerte! I'll try and have a look today or tomorrow.

File: R/readGWLdata.R
Comment (ateucher) [inline R/readGWLdata.R]: I know they were in the original, but I'm not sure we want the historical mean/max/min attached to every data frame... what about a fourth `which` option (`"hist_avg"` or something) that allows downloading just that on its own, and leave it up to the user to join them if they need to?

File: R/readGWLdata.R
Comment (ateucher) [inline R/readGWLdata.R:111]: So there is a source for well metadata coming in the next couple of weeks, if you're able to hold off then we can add that then?

File: R/gwlZypTest.R
Comment (ateucher) [inline R/gwlZypTest.R:35]: We should check the existence of columns named in `byID` and `col`. I will open an issue.

File: R/monthlyValues.R
Comment (ateucher) [inline R/monthlyValues.R:35]: Nice

File: R/readGWLdata.R
Comment (ateucher) [inline R/readGWLdata.R]: What do you think about using `which = c("all", "recent", "daily")` in the argument list, and then `which <- match.arg(which)` to check?

File: R/readGWLdata.R
Comment (ateucher) [inline R/readGWLdata.R]: See question below for whether we should always get the minMaxMean...

File: R/readGWLdata.R
Comment (ateucher) [inline R/readGWLdata.R:18]: Nice documentation

File: DESCRIPTION
Comment (ateucher) [inline DESCRIPTION]: I'm thinking we should move rgdal and sp to Suggests as they're only used for the one function (`utm_dd`). Then at the top of the body of `utm_dd` use:
```
if (!requireNamespace("rgdal") || !requireNamespace("sp")) {
  stop("You need the sp and rgdal packages installed to use this function")
}
```

Thoughts?

**Review by ateucher — CHANGES_REQUESTED** [review]
Hey @steffilazerte - this looks really great! The tests and the working vignette are fantastic!

Just a couple of small points which are open for discussion (comments inline).

The only other thing is that I generally try to avoid pipes in package functions because it makes debugging really difficult - I know the package already had pipes in it, just wondering what you think?

File: DESCRIPTION
Comment (steffilazerte) [reply DESCRIPTION]: Definitely, especially as rgdal and sp aren't lightweight packages

File: R/readGWLdata.R
Comment (steffilazerte) [reply R/readGWLdata.R]: What about an addition argument, `hist_avg = FALSE`? I've already coded it in so this way we can just wrap the code that downloads and joins in `if(hist_avg) {...}`. We can make the argument FALSE by default to minimize unnecessary downloading?

File: R/readGWLdata.R
Comment (steffilazerte) [reply R/readGWLdata.R:111]: Absolutely, no hurry. We could still push to master, if you think people may want to use it without the meta data. Or we can wait if you think it could get confusing.

File: R/readGWLdata.R
Comment (steffilazerte) [reply R/readGWLdata.R]: I have to admit it never occurred to me, but yes, that would totally work. The only small thing I don't like is that the error message is: 
```
Error in match.arg(which) : 
  'arg' should be one of “all”, “recent”, “daily”
```
Which, to an R newbie, isn't really a clear error message.

But either way, 'type' should be 'which' in the original!

**Comment by steffilazerte** [comment]
Commit 225c64e was odd. In order to keep `rgdal` and `sp` as suggests and to avoid a check error, I had to manually delete their 'import' lines from the NAMESPACE. But if I reran `devtools::document()` those import lines would be added back in... possibly a bug in `document()`?

File: R/readGWLdata.R
Comment (ateucher) [reply R/readGWLdata.R]: This is a great idea!

File: R/readGWLdata.R
Comment (ateucher) [reply R/readGWLdata.R]: I hope you don't mind, but this inspired me to write a function `arg_match` that does a better job of this. I implemented it for `get_gwl` in https://github.com/bcgov/bcgroundwater/pull/10/commits/65acc501741a1d24359be7473ee7f8445ae45c9d

Comment (steffilazerte) [reply R/readGWLdata.R]: I definitely don't mind, that's a great function! I like it much better than the original.

Comment (ateucher) [reply R/readGWLdata.R]: I think I'll make use of it in a lot of other places!

File: R/utm_dd.R
Comment (ateucher) [inline R/utm_dd.R]: I think we need `sp::CRS` here and `sp::coordinates` below

Comment (steffilazerte) [reply R/utm_dd.R]: Yup, I missed that. I also missed the @imports that called rgdal and sp, which is why I had trouble with the NAMESPACE. Fixed now in fee8dc9.

Comment (ateucher) [reply R/utm_dd.R]: 👍

File: R/gwlZypTest.R
Comment (ateucher) [reply R/gwlZypTest.R:35]: #12 - probably should look across the board to make sure it's done consistently (for later)

**Review by ateucher — APPROVED** [review]

**Pull request merged**
